packages/hub/node-tests/services/scheduled-payments/fetcher-test.ts

```diff
@@ -0,0 +1,194 @@
+import shortUUID from 'short-uuid';
+import ScheduledPaymentsFetcherService from '../../../services/scheduled-payments/fetcher';
+import { setupHub } from '../../helpers/server';
+import crypto from 'crypto';
+import { addDays, addMinutes, addMonths, subDays, subMinutes, subMonths } from 'date-fns';
+import { ScheduledPayment, ScheduledPaymentAttemptStatusEnum } from '@prisma/client';
+import { convertDateToUTC } from '../../../utils/dates';
+import { ExtendedPrismaClient } from '../../../services/prisma-manager';
+
+describe('fetching scheduled payments that are due', function () {
+  let subject: ScheduledPaymentsFetcherService;
+  let now: Date;
+  let validForDays: number;
+  let prisma: ExtendedPrismaClient;
+
+  let { getPrisma, getContainer } = setupHub(this);
+
+  this.beforeEach(async function () {
+    subject = await getContainer().lookup('scheduled-payments-fetcher');
+    now = convertDateToUTC(new Date());
+    validForDays = subject.validForDays;
+    prisma = await getPrisma();
+  });
+
+  let createScheduledPayment = async (params: any) => {
+    return prisma.scheduledPayment.create({
+      data: {
+        id: shortUUID.uuid(),
+        senderSafeAddress: '0x123',
+        moduleAddress: '0x456',
+        tokenAddress: '0x789',
+        amount: 1,
+        payeeAddress: '0xabc',
+        executionGasEstimation: 1,
+        maxGasPrice: 1,
+        feeFixedUsd: 1,
+        feePercentage: 1,
+        nonce: 'abc',
+        payAt: params.payAt,
+        recurringDayOfMonth: params.recurringDayOfMonth,
+        recurringUntil: params.recurringUntil,
+        validForDays: params.validForDays,
+        spHash: crypto.randomBytes(20).toString('hex'),
```

Comment:
does the hub care what the spHash is? I thought this was something that was computed on-chain based on the parameters to contract method that the crank calls to make a payment--where the contract stores the spHash and the subsequent calls to make a payment must hash to the spHash that the contract stored

Reply:
I think the hub doesn't care about it, it just has it saved so that it can use it to execute payments. The idea is that the client is the one who will form the `spHash` and send it to the hub which will save it along with all the params and attempt the contract call to schedule a payment. It will then use the `spHash` later when trying to execute the payment.

Reply:
We may want to have the hub verify that the spHash it has is present in the module state. Otherwise, we should likely assume that a payment was either made or canceled out-of-band.

Reply:
Good point. I would put that check in the service that will execute the scheduled payment.